tests/components/element-editor/validation/elements/contact/contact-validation-child.spec.tsx

```diff
@@ -0,0 +1,102 @@
+import { render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { getValidationOptionsByType } from "src/components/element-editor/validation";
+import { ContactValidationChild } from "src/components/element-editor/validation/elements";
+import { EElementType, EValidationType } from "src/context-providers";
+import { TestHelper } from "src/util/test-helper";
+
+describe("ContactValidationChild", () => {
+    beforeEach(() => {
+        global.ResizeObserver = jest.fn().mockImplementation(() => ({
+            observe: jest.fn(),
+            unobserve: jest.fn(),
+            disconnect: jest.fn(),
+        }));
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+
+    it("should render ContactValidationChild component with default type and error message", () => {
+        renderComponent({
+            formContext: {
+                defaultValues: {
+                    validation: [
+                        {
+                            validationType: EValidationType.CONTACT_NUMBER,
+                            validationErrorMessage: "Invalid contact number.",
+                        },
+                    ],
+                },
+            },
+        });
+        const validationType = screen.getByTestId("validation-type");
+        expect(validationType).toBeInTheDocument();
+        expect(validationType).toHaveTextContent("Contact number format");
+        const validationErrorMessage = screen.getByTestId(
+            "validation-error-message"
+        );
+        expect(validationErrorMessage).toBeInTheDocument();
+    });
+
+    it("should render ContactValidationRule when defaultCountryCode is '65' and displayAsFixedCountryCode is true", () => {
+        renderComponent({
+            formContext: {
+                defaultValues: {
+                    defaultCountryCode: "65",
+                    displayAsFixedCountryCode: true,
+                },
+            },
+        });
+        expect(
+            screen.queryByTestId("contact-validation-rule")
```

Comment:
though this assertion is valid, it's still better to use `getBy` to test for the absence of an element because `queryBy` will not throw an error.

```suggestion
            screen.getByTestId("contact-validation-rule")
```